### PR TITLE
Enemy spawner

### LINKED ISSUE #112: Enemy spawner
Ideas pls

### PR DESCRIPTION
**Pull request by jeffsieu**
image

Enemy spawner spawns 5 enemies per room.

At first, only the starting room spawns enemies. When enemies in a room are killed, it will Activate() all adjacent rooms, causing them to spawn enemies.

Also:
- Add signal `died` to HealthEntity; used by EnemySpawner to detect enemy death
- Wraps HealthBar nodes within a Node2D to set their z-index, so that HealthBars appear underneath the main GUI

Resolves #112

### CODE REVIEW

**Review by daniellimws — APPROVED** [review]
Nicesah!

File: Scripts/GUI.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+using System;
+using Godot;
+
+public class GUI : Control
```
Comment (owenl131) [inline Scripts/GUI.cs:4]: Wanna name this something more specific?

**Review by owenl131 — APPROVED** [review]
Ok so far

File: Scripts/GUI.cs (hunk 1 above)
Comment (jeffsieu) [reply Scripts/GUI.cs:4]: Like what? I was planning to make it just some helper class that can be used to access GUI elements from other Nodes.

**Pull request merged**